core/src/main/kotlin/fr/sncf/osrd/api/api_v2/conflicts/ConflictDetectionEndpointV2.kt

```diff
@@ -0,0 +1,95 @@
+package fr.sncf.osrd.api.api_v2.conflicts
+
+import fr.sncf.osrd.api.ConflictDetectionEndpoint.ConflictDetectionResult
+import fr.sncf.osrd.api.ExceptionHandler
+import fr.sncf.osrd.conflicts.TrainRequirements
+import fr.sncf.osrd.conflicts.detectConflicts
+import fr.sncf.osrd.standalone_sim.result.ResultTrain
+import fr.sncf.osrd.utils.units.TimeDelta
+import java.time.Duration.between
+import java.time.ZonedDateTime
+import org.takes.Request
+import org.takes.Response
+import org.takes.Take
+import org.takes.rq.RqPrint
+import org.takes.rs.RsJson
+import org.takes.rs.RsText
+import org.takes.rs.RsWithBody
+import org.takes.rs.RsWithStatus
+
+class ConflictDetectionEndpointV2 : Take {
+    override fun act(req: Request?): Response {
+        return try {
+            val body = RqPrint(req).printBody()
+            val request =
+                conflictRequestAdapter.fromJson(body)
+                    ?: return RsWithStatus(RsText("missing request body"), 400)
+
+            val minStartTime = request.trainRequirements.values.minBy { it.startTime }.startTime
+            val trainRequirements = makeTrainRequirements(request.trainRequirements, minStartTime)
+            val conflicts = detectConflicts(trainRequirements)
+            val res = makeConflictDetectionResponse(conflicts, minStartTime)
+
+            RsJson(RsWithBody(conflictResponseAdapter.toJson(res)))
+        } catch (ex: Throwable) {
+            ExceptionHandler.handle(ex)
+        }
+    }
+}
+
+private fun makeTrainRequirements(
+    trainReqs: Map<Long, TrainRequirementsRequest>,
+    startTime: ZonedDateTime
+): List<TrainRequirements> {
+    val trainRequirements = mutableListOf<TrainRequirements>()
+    for ((id, trainReq) in trainReqs) {
+        val delta = TimeDelta(between(trainReq.startTime, startTime).toMillis())
```

Comment:
I suspect that the math is off: `between(start, end)` returns a negative value if `end < start`. Considering `startTime` is the earliest value, `delta` would be `<=0`. Which is not necessarily a problem in itself, but I don't think *adding* that negative delta would offset the times in the correct way.